src/credentials/vue/authForms/manageBuilderId.vue

```diff
@@ -0,0 +1,164 @@
+                <div class="form-section">
+                    <div v-on:click="signout()" style="cursor: pointer; color: #75beff">Sign out</div>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+<script lang="ts">
+import { PropType, defineComponent } from 'vue'
+import BaseAuthForm from './baseAuth.vue'
+import FormTitle from './formTitle.vue'
+import { AuthFormId, AuthStatus } from './shared.vue'
+import { WebviewClientFactory } from '../../../webviews/client'
+import { AuthWebview } from '../show'
+
+const client = WebviewClientFactory.create<AuthWebview>()
+
+export default defineComponent({
+    name: 'CredentialsForm',
+    extends: BaseAuthForm,
+    components: { FormTitle },
+    props: {
+        state: {
+            type: Object as PropType<BaseBuilderIdState>,
+            required: true,
+        },
+    },
+    data() {
+        return {
+            stage: stages.START as BuilderIdStage,
+            isConnected: false,
+            builderIdCode: '',
+            canShowAll: false,
+        }
+    },
+    async created() {
+        await this.update()
+        this.canShowAll = true
+    },
+    methods: {
+        async startSignIn() {
+            this.stage = stages.WAITING_ON_USER
+            await this.state.startBuilderIdSetup()
+            await this.update()
+        },
+        async update() {
+            this.stage = await this.state.stage()
+            this.isConnected = await this.state.isAuthConnected()
+            this.emitAuthConnectionUpdated(this.state.id)
+        },
+        async signout() {
+            await this.state.signout()
+
+            this.update()
+        },
+    },
+})
+
+export const stages = {
+    START: 'START',
+    WAITING_ON_USER: 'WAITING_ON_USER',
+    CONNECTED: 'CONNECTED',
+} as const
+
+type BuilderIdStage = (typeof stages)[keyof typeof stages]
+
+/**
+ * Manages the state of Builder ID.
+ */
+abstract class BaseBuilderIdState implements AuthStatus {
+    protected _stage: BuilderIdStage = stages.START
+
+    get id(): AuthFormId {
+        throw new Error('Needs implementation')
+    }
+
+    protected _startBuilderIdSetup(): Promise<void> {
+        throw new Error('Needs implementation')
+    }
+
+    async isAuthConnected(): Promise<boolean> {
+        throw new Error('Needs implementation')
+    }
+
+    async startBuilderIdSetup(): Promise<void> {
+        this._stage = stages.WAITING_ON_USER
+        return this._startBuilderIdSetup()
+    }
+
+    async stage(): Promise<BuilderIdStage> {
+        const isAuthConnected = await this.isAuthConnected()
+        this._stage = isAuthConnected ? stages.CONNECTED : stages.START
+        return this._stage
+    }
+
+    async signout(): Promise<void> {
+        await client.signoutBuilderId()
+    }
+}
+
+export class CodeWhispererBuilderIdState extends BaseBuilderIdState {
+    override get id(): AuthFormId {
+        return 'BUILDER_ID_CODE_WHISPERER'
+    }
```

Comment:
Is this ever used?

Reply:
I thought I did at first, but don't actually need it now. Added it to my TODO since my other feature branches also did this.